brainscore/model_interface.py

```diff
@@ -117,6 +117,31 @@ class Task:
                   - choice        (choice) object 'dog' 'cat' 'chair' 'flower' 'plane'
         """
 
+        odd_one_out = 'odd_one_out'
+        """
+        Predict the odd-one-out elements for a list of triplets of stimuli. Calling `start_task(...)` sets the 
+        similarity measure to use for the odd-one-out task. Calling `look_at(...)` with a list of triplets of stimuli
+        will compute the similarity matrix for all pairs of stimuli and compute the odd one out for each triplet.  
+        
+        Output a :class:`~brainio.assemblies.BehavioralAssembly` with the stimulus_ids as the value.
+        
+        Example:
+
+        Setting up a odd-one-out task for a list of triplets with 
+        `start_task(BrainModel.Task.odd_one_out, [image_1, image_2, image_3], similarity_measure)`
+        and calling `look_at(...)` could output 
```

Comment:
No you need to pass `data = [images, triplets]`. We compute all activations and then we do the odd one out task for each triplet. So we get `len(triplets)` choices.

Reply:
ok maybe you can update the documentation, I'm not sure I get what the format of `triplets` is (a list of triplets where each entry is a certain `stimulus_id`?)

Passing an additional parameter beyond `stimuli` (or `images`) is currently not supported by the interface, this would be quite some refactor. So I wonder if the "list of triplet stimuli" format I suggested above wouldn't be easier? In the model, we could still compute activations on the unique set of stimuli but then build the triplets matrix from the stimuli parameter

Reply:
The format of triplets is a list of lists `triplet_i = [stimulus_reference_i1, stimulus_reference_i2, stimulus_reference_i3]`, where the `stimulus_reference_ij`s are integers pointing to a stimulus.

Reply:
I understand that we can make it work, but the strong preference is to keep the interface between the different tasks as unified as possible. What you wrote is exactly right, i.e. something like
```
# in the benchmark
def __call__(self, model: BrainModel) -> Score:
    model.start_task(BrainModel.Task.odd_one_out, fitting_stimuli=...)
    model_choices = model.look_at(self.stimuli)
    score = self.metric(model_choices, self.assembly)
    return score
```
```
# in model-tools OddOneOut decoder
def look_at(stimuli):
    unique_stimuli = np.unique(stimuli)
    features = self.activations_model(unique_stimuli, layers=self.readout)
    triplets = self.triplets_from_stimuli(stimuli)
    similarities = self.similarity_measure(features, triplets)
    choices = ....
    return choices
```

Reply:
Ok, I will happily change it in `model-tools` once we finalize this PR.